packages/react/src/ResourceArrayInput/ResourceArrayInput.utils.ts

```diff
@@ -1,41 +1,62 @@
 import {
-  InternalTypeSchema,
-  SliceDefinition,
+  InternalSchemaElement,
+  MedplumClient,
   SliceDiscriminator,
   SlicingRules,
   TypedValue,
   arrayify,
-  getElementDefinitionFromElements,
-  getTypedPropertyValueWithSchema,
+  getNestedProperty,
+  isEmpty,
   isPopulated,
   matchDiscriminant,
+  tryGetProfile,
 } from '@medplum/core';
+import { SupportedSliceDefinition, isSupportedSliceDefinition } from '../SliceInput/SliceInput.utils';
 
 function isDiscriminatorComponentMatch(
   typedValue: TypedValue,
   discriminator: SliceDiscriminator,
-  slice: SupportedSliceDefinition
+  slice: SupportedSliceDefinition,
+  profileUrl: string | undefined
 ): boolean {
-  for (const elementList of [slice.elements, slice.typeSchema?.elements]) {
-    let nestedProp: TypedValue | TypedValue[] | undefined;
-    if (isPopulated(elementList)) {
-      const ed = getElementDefinitionFromElements(elementList, discriminator.path);
-      if (ed) {
-        nestedProp = getTypedPropertyValueWithSchema(typedValue.value, discriminator.path, ed);
-      }
-    }
+  const elementList = slice.typeSchema?.elements ?? slice.elements;
 
-    if (nestedProp) {
-      return arrayify(nestedProp)?.some((v: any) => matchDiscriminant(v, discriminator, slice, elementList)) ?? false;
-    }
+  // const pathParts = discriminator.path.split('.');
+  // let lastEd: InternalSchemaElement | undefined;
+  // let lastNestedProp: TypedValue | TypedValue[] | undefined;
+  // for (let i = 0; i < pathParts.length; i++) {
+  //   const pathPart = pathParts[i];
+  //   const pathSoFar = pathParts.slice(0, i+1).join('.')
+  //   lastEd = getElementDefinitionFromElements(elementList, pathSoFar);
+  //   if (lastEd) {
+  //     lastNestedProp = getTypedPropertyValueWithSchema(typedValue, discriminator.path, lastEd);
+  //   }
+  // }
+  // nestedProp = lastNestedProp;
+
+  // let nestedProp: TypedValue | TypedValue[] | undefined;
+  // const ed = getElementDefinitionFromElements(elementList, discriminator.path);
+  // if (ed) {
+  // nestedProp = getTypedPropertyValueWithSchema(typedValue, discriminator.path, ed);
+  // }
+
+  const nestedProp = getNestedProperty(typedValue, discriminator.path, profileUrl);
+
+  if (nestedProp) {
+    const result =
+      arrayify(nestedProp)?.some((v: any) => matchDiscriminant(v, discriminator, slice, elementList)) ?? false;
+    return result;
+  } else {
+    console.assert(false, 'getNestedProperty[%s] in isDiscriminatorComponentMatch missed', discriminator.path);
```

Comment:
Is this a pattern we want to use in the React components for error reporting?  Seems pretty handy